src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -941,10 +941,945 @@ class spell_silverpine_forsaken_trooper_masterscript_high_command : public Spell
+
+    PATH_DARIUS_ENTER                       = 448830,
+    PATH_IVAR_ENTER                         = 448840,
+    PATH_IVAR_RANE_01                       = 448841,
+    PATH_IVAR_RANE_02                       = 448842,
+    PATH_IVAR_RANE_03                       = 448843,
+    PATH_IVAR_EXIT                          = 448844,
+    PATH_DARIUS_EXIT                        = 448831
+};
+
+// Armoire - 44893
+struct npc_silverpine_armoire : public VehicleAI
+{
+    npc_silverpine_armoire(Creature* creature) : VehicleAI(creature) { }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+        {
+            if (player->GetQuestStatus(QUEST_WAITING_TO_EXSANGUINATE) == QUEST_STATUS_INCOMPLETE)
+				_playerGUID = player->GetGUID();
+        }
+    }
+
+    void PassengerBoarded(Unit* passenger, int8 /*seatId*/, bool apply) override
+    {
+        if (apply)
+        {
+            if (Player* player = passenger->ToPlayer())
+            {
+                if (player->GetQuestStatus(QUEST_WAITING_TO_EXSANGUINATE) == QUEST_STATUS_INCOMPLETE)
+                    _events.ScheduleEvent(EVENT_START_SCENE_EXSANGUINATE, 1s);
+            }
+        }
+        else
+        {
+            if (Creature* darius = ObjectAccessor::GetCreature(*me, _crowleyGUID))
+                darius->DespawnOrUnsummon();
+
+            if (Creature* ivar = ObjectAccessor::GetCreature(*me, _bloodfangGUID))
+                ivar->DespawnOrUnsummon();
+
+            if (Creature* rane = ObjectAccessor::GetCreature(*me, _raneGUID))
+                rane->DespawnOrUnsummon();
+
+            me->DespawnOrUnsummon(1s);
+        }
+    }
+
+    void SetGUID(ObjectGuid const& guid, int32 id) override
+    {
+        switch (id)
+        {
+            case NPC_DEATHSTALKER_RANE_YORICK:
+                _raneGUID = guid;
+                break;
+
+            case NPC_LORD_DARIUS_CROWLEY:
+                _crowleyGUID = guid;
+                break;
+
+            case NPC_PACKLEADER_IVAR_BLOODFANG:
+                _bloodfangGUID = guid;
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_START_SCENE_EXSANGUINATE:
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    {
+                        player->CastSpell(player, SPELL_SUMMON_CROWLEY_BLOODFANG_MASTER, true);
+
+                        _events.ScheduleEvent(EVENT_START_SCENE_EXSANGUINATE + 1, 250ms);
+                    }
+                    break;
+                }
+
+                case EVENT_START_SCENE_EXSANGUINATE + 1:
+                {
+                    if (Creature* darius = ObjectAccessor::GetCreature(*me, _crowleyGUID))
```

Comment:
name consistency (that counts for the entire UpdateAI())

Reply:
Feedback applied!